algosdk/dryrun_results.py

```diff
@@ -0,0 +1,227 @@
+import base64
+from typing import List
+import tabulate as tlib
+from tabulate import tabulate, TableFormat, DataRow
+
+
+tlib.MIN_PADDING = 0
+
+
+class StackPrinterConfig:
+    DEFAULT_MAX_WIDTH: int = 30
+
+    def __init__(self, max_width=DEFAULT_MAX_WIDTH, top_of_stack_first=True):
+        self.max_width = max_width
+        self.top_of_stack_first = top_of_stack_first
+
+
+class DryrunResponse:
+    def __init__(self, drrjson: dict):
+        for param in ["error", "protocol-version", "txns"]:
+            assert (
+                param in drrjson
+            ), f"expecting dryrun response object to have key '{param}' but it is missing"
+
+        # These are all required fields
+        self.error = drrjson["error"]
+        self.protocol = drrjson["protocol-version"]
+        self.txns = [DryrunTransactionResult(txn) for txn in drrjson["txns"]]
+
+
+class DryrunTransactionResult:
+    def __init__(self, dr):
+        assert (
+            "disassembly" in dr
+        ), "expecting dryrun transaction result to have key 'disassembly' but its missing"
+
+        self.disassembly = dr["disassembly"]
+
+        optionals = [
+            "app-call-messages",
+            "local-deltas",
+            "global-delta",
+            "cost",
+            "logic-sig-messages",
+            "logic-sig-disassembly",
+            "logs",
+        ]
+        for field in optionals:
+            if field in dr:
+                setattr(self, field.replace("-", "_"), dr[field])
+            else:
+                setattr(self, field.replace("-", "_"), None)
+
+        traces = ["app-call-trace", "logic-sig-trace"]
+        for trace_field in traces:
+            if trace_field in dr:
+                setattr(
+                    self,
+                    trace_field.replace("-", "_"),
```

Comment:
```suggestion
                    attrname(trace_field)
```

assuming you added `attrname()`